### ISSUE TITLE
Test failures: TestDateTime.FileTimeOperators and TestDateTime.SetFromUTCDateTime

### ISSUE THREAD

**Issue by candrews** [xbmc/xbmc#18299]
### Describe the bug
Here is a clear and concise description of what the problem is:

These two tests are failing for Kodi 19a1:
TestDateTime.FileTimeOperators
TestDateTime.SetFromUTCDateTime

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Tests should pass :)

## Actual Behavior

The tests fail. Here are some excerpts from LastTest.log:
```
444/547 Testing: TestDateTime.FileTimeOperators
444/547 Test: TestDateTime.FileTimeOperators
Command: "/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/kodi-19.0_alpha1_build/kodi-test" "--gtest_filter=TestDateTime.FileTimeOperators"
Directory: /var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/kodi-19.0_alpha1_build
"TestDateTime.FileTimeOperators" start time: Aug 17 06:48 CEST
Output:
----------------------------------------------------------
Note: Google Test filter = TestDateTime.FileTimeOperators
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from TestDateTime
[ RUN      ] TestDateTime.FileTimeOperators
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:43: Failure
Value of: dateTime1 < fileTime
  Actual: false
Expected: true
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:44: Failure
Value of: dateTime1 > fileTime
  Actual: true
Expected: false
[  FAILED  ] TestDateTime.FileTimeOperators (0 ms)
[----------] 1 test from TestDateTime (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (49 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestDateTime.FileTimeOperators

 1 FAILED TEST
Clean shutdown of TestGlobalPattern1
<end of output>
```
```
458/547 Testing: TestDateTime.SetFromUTCDateTime
458/547 Test: TestDateTime.SetFromUTCDateTime
Command: "/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/kodi-19.0_alpha1_build/kodi-test" "--gtest_filter=TestDateTime.SetFromUTCDateTime"
Directory: /var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/kodi-19.0_alpha1_build
"TestDateTime.SetFromUTCDateTime" start time: Aug 17 06:48 CEST
Output:
----------------------------------------------------------
Note: Google Test filter = TestDateTime.SetFromUTCDateTime
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from TestDateTime
[ RUN      ] TestDateTime.SetFromUTCDateTime
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:250: Failure
Expected equality of these values:
  dateTime3.GetYear()
    Which is: 1922
  1991
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:251: Failure
Expected equality of these values:
  dateTime3.GetMonth()
    Which is: 11
  5
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:253: Failure
Expected equality of these values:
  dateTime3.GetHour()
    Which is: 19
  12
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:254: Failure
Expected equality of these values:
  dateTime3.GetMinute()
    Which is: 36
  34
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:255: Failure
Expected equality of these values:
  dateTime3.GetSecond()
    Which is: 26
  56
[  FAILED  ] TestDateTime.SetFromUTCDateTime (0 ms)
[----------] 1 test from TestDateTime (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (48 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestDateTime.SetFromUTCDateTime

 1 FAILED TEST
Clean shutdown of TestGlobalPattern1
<end of output>
```
Full `LastLog.log`: https://737544.bugs.gentoo.org/attachment.cgi?id=655024
build output: https://bugs.gentoo.org/attachment.cgi?id=655018

### To Reproduce

1. Run the Kodi 19a1 tests

### Debuglog

The debuglog can be found here:
n/a

### Screenshots 
Here are some links or screenshots to help explain the problem:

n/a

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

Originally reported at https://bugs.gentoo.org/737544

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [X] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Gentoo/Linux
 - Kodi version: 19a1

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by candrews** [comment]
Based on version control history, I think @lrusak would know the most about this issue.

**Comment by lrusak** [comment]
Not sure, it's likely something screwy with your time zone.

Really the CDateTime code needs to be fixed. It's still something I want to work on for Kodi 19. The tests really just point out these issues.

**Comment by basilgello** [comment]
> Not sure, it's likely something screwy with your time zone.
> 
> Really the CDateTime code needs to be fixed. It's still something I want to work on for Kodi 19. The tests really just point out these issues.

I encountered the same issue on Debian sid, and now building the fix for both tests. Once all tests are successfully passed, I PR it.

Briefly, the **TestDateTime.SetFromUTCDateTime** fails due to usage of ULARGE_INTEGER instead of LARGE_INTEGER in CDateTimeSpan and CDateTime implementations. The **TestDateTime.FileTimeOperators** test fails because the test logic does not account timezone additions and subtraction, and the problem does not manifest on `TZ=UTC+{00-13}:00` but does manifest on `TZ=UTC-{01,13}:00`. Typically, Debian sbuild operates at TZ=UTC so I did not catch this issue before.
